src/serac/numerics/mesh_utils.cpp

```diff
@@ -172,20 +173,59 @@ namespace mesh {
 void InputOptions::defineInputFileSchema(axom::inlet::Table& table)
 {
   // mesh path
-  table.addString("mesh", "Path to Mesh file").required();
+  table.addString("mesh", "Path to Mesh file");
 
   // Refinement levels
   table.addInt("ser_ref_levels", "Number of times to refine the mesh uniformly in serial.").defaultValue(0);
   table.addInt("par_ref_levels", "Number of times to refine the mesh uniformly in parallel.").defaultValue(0);
+
+  // mesh generation options
+  auto& elements = table.addTable("elements");
+  // JW: Can these be specified as requierd if elements is defined?
+  elements.addInt("x", "x-dimension");
```

Comment:
Did you try  `elements.addInt("x", "x-dimension").required(true);"

Otherwise you can add a custom verifier via a lambda given to the `elements` table.

Reply:
Example of verifier, that is actually pretty close to your use case:

https://github.com/LLNL/axom/blob/494fc6429ac217b9d6aff5c9bc967dd3ee8aa5b9/src/axom/inlet/examples/verification.cpp#L32-L53

Reply:
Yes, when I added `.required()` to "x", it required it even when `elements` was not defined (e.g. in the case when we're building a mesh from file).

I will try the custom verifier.

Reply:
I wonder if we could support this by setting the table to not be required to the table the the field being required.  Something like:

```
elements = table.addTable("elements").required(false);
elements.addInt("x").required(true);
```

This may be a bad idea...

Reply:
I've started to look into this, and recalled a relevant comment from a recent inlet PR:
https://github.com/LLNL/axom/pull/427#discussion_r556825102

I think the idea of only checking sub-objects when the parent object is defined makes sense, but it might be a little tricky to determine when the parent table is defined.  Currently this is based on the existence of at least one expected subobject, so `elements = { q = 1 }` would not count as "defined", but `elements = { x = 1 }` would.  

I suppose we could add logic to just check if the `elements` table is a valid "path" in the input file and go from there.